### PR TITLE
PR - Issue 568 typecast fix for 1D array or list

### PR DESCRIPTION
**Pull request by nirmalchandra**
# <Feature Title>

PR Branch
**_#ADD LINK TO THE PR BRANCH_**

## Checklist:

* [x] Unit tests added

* [x] Integration tests added

* [x] Test names are meaningful

* [x] Feature manually tested

* [x] Branch build passed

* [x] No 'package.*' in the imports

* [ ] Relevant Wiki page updated with clear instruction for the end user
  * [x] Not applicable. This was only a refactor change, no functional or behaviour changes were introduced

* [ ] Http test added to `http-testing` module(if applicable) ?
  * [x] Not applicable. The changes did not affect HTTP automation flow

* [ ] Kafka test added to `kafka-testing` module(if applicable) ?
  * [x] Not applicable. The changes did not affect Kafka automation flow

### CODE REVIEW

**Review by authorjapps — APPROVED** [review]

File: core/src/main/java/org/jsmart/zerocode/core/utils/FieldTypeConversionUtils.java [hunk 1]
```diff
@@ -88,6 +88,21 @@ private static void replaceNodeValue(Map.Entry<String, Object> entry, Object thi
             LOGGER.error(errorMsg + "\nException Details:" + exx);
```
Comment (M3lkior) [inline core/src/main/java/org/jsmart/zerocode/core/utils/FieldTypeConversionUtils.java:88]:
```suggestion
            LOGGER.error("{} with exception details: {}",  errorMsg, exx);
```

File: core/src/test/java/org/jsmart/zerocode/core/utils/FieldTypeConversionUtilsTest.java [hunk 2]
```diff
@@ -87,14 +91,17 @@ public void testSubstituted_v4() throws IOException {
...
 
         JsonNode jsonNode = mapper.valueToTree(stepMap);
 
         assertEquals(true, jsonNode.get("found").asBoolean());
         assertEquals("{\"id\":2.35,\"name\":\"Bar - 2.35\",\"isActive\":false,\"longField\":1569683094000}",
                 jsonNode.get("results").get(1).toString());
         assertEquals("address line1", jsonNode.get("currentAddress").get("line1").asText());
+        assertEquals(jsonNode.get("ids").get(0).asInt(), jsonNode.get("results").get(0).get("id").asInt());
+        assertEquals(1, jsonNode.get("ids").get(0).asInt() );
+        assertEquals(2, jsonNode.get("ids").get(1).asInt());
```
Comment (M3lkior) [inline core/src/test/java/org/jsmart/zerocode/core/utils/FieldTypeConversionUtilsTest.java:104]: int or float ? ;)

**Review by M3lkior — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/jsmart/zerocode/core/utils/FieldTypeConversionUtilsTest.java (hunk 2 above)
Comment (authorjapps) [reply core/src/test/java/org/jsmart/zerocode/core/utils/FieldTypeConversionUtilsTest.java]: Fixed

**Review by M3lkior — CHANGES_REQUESTED** [review]

**Comment by authorjapps** [comment]
Thanks for reviewing @M3lkior.
Both review comments have been fixed.
Will be merging the PR and releasing soon.
ETA: 11/June

**Pull request merged**